src/serac/coefficients/coefficient_extensions.hpp

```diff
@@ -14,41 +14,151 @@
 
 #include <functional>
 #include <memory>
-
+#include <variant>
+#include <vector>
 #include "mfem.hpp"
 
 #include "serac/numerics/expr_template_ops.hpp"
-
+#include "serac/physics/utilities/finite_element_state.hpp"
 /**
  * @brief Functionality that extends current MFEM capabilities
  *
  */
 namespace serac::mfem_ext {
 
+namespace detail {
+// methods for determining index type
+template <typename T>
+struct index_t {
+  using type = std::size_t;
+};
+
+template <typename T>
+struct index_t<mfem::Array<T>> {
+  using type = int;
+};
+
+// methods for determining the size of a container
+template <typename T>
+int size(T container)
+{
+  return static_cast<int>(container.size());
```

Comment:
You are probably losing precision here.  Most stl `size()` calls return `size_t` which is normally `unsigned int`

Reply:
@white238 You're absolutely right. I should probably add a comment, but the original reason why I moved to int is because mfem's size container returns `int`. I can try moving to `auto` for the std collections though.

Reply:
you can promote int to unsigned in safely so they are both the same..

Reply:
That's true, but if I use implicit conversions I get the following error:
`error: conversion from 'std::vector<int>::size_type' {aka 'long unsigned int'} to 'int' may change value [-Werror=conversion]`